src/components/attendee/NavOverlay/SearchOverlay/SearchOverlay.tsx

```diff
@@ -0,0 +1,147 @@
+import { ChangeEvent, useCallback, useMemo, useState } from "react";
+
+import {
+  COVERT_ROOM_TYPES,
+  DEFAULT_PARTY_NAME,
+  PERSON_TAXON,
+  SPACE_TAXON,
+} from "settings";
+
+import { AlgoliaSearchIndex } from "types/algolia";
+import { Room } from "types/rooms";
+
+import { isDefined } from "utils/types";
+
+import { useAlgoliaSearch } from "hooks/algolia/useAlgoliaSearch";
+import { useSpaceParams } from "hooks/spaces/useSpaceParams";
+import { useWorldAndSpaceBySlug } from "hooks/spaces/useWorldAndSpaceBySlug";
+import { useDebounceSearch } from "hooks/useDebounceSearch";
+import { useRelatedVenues } from "hooks/useRelatedVenues";
+
+import { Loading } from "components/molecules/Loading";
+
+import { InputField } from "components/atoms/InputField";
+
+import { PortalItem } from "./PortalItem/PortalItem";
+
+import CN from "./SearchOverlay.module.scss";
+
+type SearchOverlayProps = {
+  onClose: () => void;
+};
+export const SearchOverlay: React.FC<SearchOverlayProps> = ({ onClose }) => {
+  const { worldSlug, spaceSlug } = useSpaceParams();
+  const { space, worldId } = useWorldAndSpaceBySlug(worldSlug, spaceSlug);
+  const [searchValue, setSearchValue] = useState("");
+
+  const { searchQuery, setSearchInputValue } = useDebounceSearch();
+
+  const onSearchInputChange = useCallback(
+    (e: ChangeEvent<HTMLInputElement>) => {
+      setSearchValue(e.target.value);
+    },
+    [setSearchValue]
+  );
+
+  const initiateSearch = () => {
+    setSearchInputValue(searchValue);
+  };
+
+  const { isLoading, relatedVenues } = useRelatedVenues();
+
+  const enabledRelatedPortals = useMemo<Room[]>(
+    () =>
+      relatedVenues
+        .flatMap((venue) => venue.rooms ?? [])
+        .filter((portal) => {
+          if (
+            isDefined(portal.type) &&
+            COVERT_ROOM_TYPES.includes(portal.type)
+          ) {
+            return false;
+          }
+
+          return portal.isEnabled;
+        }),
+    [relatedVenues]
+  );
+
+  const foundPortals = useMemo<JSX.Element[]>(() => {
+    if (!searchQuery) return [];
```

Comment:
is there a possibility of re-render? in that case 
```suggestion
    if (!searchQuery) return ALWAYS_EMPTY_ARRAY;
```